### DISCUSSION TITLE
Allow access to internal format for API usage

### DISCUSSION THREAD

**Discussion by IxDay** [FiloSottile/age#223] [General]
## What were you trying to do

I am building an ssh agent able to decrypt age files using the extension mechanism based on [previous work](https://github.com/42wim/sagent) from [42wim](https://github.com/42wim). In the code I am parsing the age header to detect what ssh key could be used for decryption (see code [here](https://github.com/IxDay/janus/blob/master/pkg/janus/agent.go#L175))

## What happened

The format code is stored in the internal directory making it not available externally. This forced me to copy the entire file in my repository. Would it be possible to move this to a globally available namespace, allowing program and libraries to parse the header?

**Comment by str4d** [comment]
The way in which agents will be supported by `age` and `rage` is the plugin system (I've recently merged plugin support into `rage` behind the `unstable` feature flag). At decryption time, a plugin is given every stanza in the header, and will be making its decisions there. So once `age` has plugin support, I expect there will be an exposed way to parse stanzas, without needing to expose parsing of the entire header.

**Comment by FiloSottile** [comment]
This is a spot-on use case for plugins, which we'll hopefully publish the spec for very soon!
